File: backend/src/api_schema/base_stats.go
package api_schema

type BaseStats struct {
	Id             int     `json:"id"`
	HitPoints      int     `json:"hit_points"`
	Attack         int     `json:"attack"`
	Defense        int     `json:"defense"`
	SpecialAttack  int     `json:"special_attack"`
	SpecialDefense int     `json:"special_defense"`
	Speed          int     `json:"speed"`
	Average        float64 `json:"average"`
	Total          int     `json:"total"`
}

File: frontend/src/entity/BaseStats.ts
export interface BaseStats {
    id: number;
    hitPoints: number;
    attack: number;
    defense: number;
    specialAttack: number;
    specialDefense: number;
    speed: number;
    average: number;
    total: number;
}

export const NewBaseStatsFromJson = (baseStats: any) : BaseStats => {
    return {
        id: baseStats.id,
        hitPoints: baseStats.hit_points,
        attack: baseStats.attack,
        defense: baseStats.defense,
        specialAttack: baseStats.special_attack,
        specialDefense: baseStats.special_defense,
        speed: baseStats.speed,
        average: baseStats.average,
        total: baseStats.total,
    };
}